TheMovieDatabase/AppDelegate.swift

```diff
@@ -57,3 +58,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         window?.rootViewController = tabBarController
     }
 }
+
+// MARK: - UITabBarControllerDelegate
+
+extension AppDelegate: UITabBarControllerDelegate {
+
+    func tabBarController(_ tabBarController: UITabBarController,
+                          shouldSelect viewController: UIViewController) -> Bool {
+        let vc = viewController as? UINavigationController
+        let array = vc?.viewControllers
+        if tabBarController.selectedViewController === viewController && array?.count == 1 {
```

Comment:
You can check in place without creating an array: `vc?.viewControllers.count == 1`

Reply:
sure